examples/plot_mdm_spd_helper.py

```diff
@@ -0,0 +1,145 @@
+import numpy
+import matplotlib
+import matplotlib.pyplot as plt
+import matplotlib.gridspec as gridspec
+
+plt.style.use('seaborn')
+
+fontsize=15
+matplotlib.rc('font', size=fontsize)
+matplotlib.rc('text', usetex=True)
+matplotlib.rc('legend', fontsize=fontsize)
+matplotlib.rc('font',
+              family='times',
+              serif=['Computer Modern Roman'],
+              monospace=['Computer Modern Typewriter'])
+
+EPS=1e-8
+
+class DatasetSPD_2D():
+
+    def __init__(self,n_samples,n_features,n_classes):
+        self.n_samples=n_samples
+        self.n_features=n_features
+        self.n_classes=n_classes
+        self.data_helper=DataHelper()
+
+    def generate_sample_dataset(self):
+        X,Y=self.easy_gauss()
+        X,Y=self.data_helper.shuffle(X,Y)
+        return X,Y
+
+    def mat2vec(self,mat):
+        n=mat.shape[-1]
+        return mat[numpy.tril_indices(n)]
+
+    # returns one random rotation of dimension n
+    def random_rotation(self):
+        if(self.n_features==2):
+            theta=numpy.random.uniform(0,2*numpy.pi)
+            c,s=numpy.cos(theta),numpy.sin(theta)
+            s=numpy.array([[c,-s],[s,c]])
+            return s
+        z=(numpy.random.randn(n,n)+1j*numpy.random.randn(n,n))/numpy.sqrt(2.0)
+        q,r=numpy.linalg.qr(z)
+        d=numpy.diag(r)
+        ph=numpy.diag(d/numpy.abs(d))
+        # q=numpy.dot(q,numpy.dot(ph,q))
+        s=numpy.dot(q,ph)
+        return s
+
+    # returns N random rotation matrices close of var_rot around R
+    def random_rotations(self,R,var_rot):
+        rots=numpy.zeros((self.n_samples,self.n_features,self.n_features))
+        for i in range(self.n_samples):
+            rots[i]=numpy.dot(R,self.divide_angle_of_cov2(self.random_rotation(),var_rot))
+        return rots
+
+    def angle_of_rot2(self,r):
+        return numpy.arctan(r[0][1]/r[0][0])
+
+    def divide_angle_of_cov2(self,r,alpha):
+        angle=self.angle_of_rot2(r)*alpha
+        c,s=numpy.cos(angle),numpy.sin(angle)
+        return numpy.array([[c,-s],[s,c]])
+
+    def easy_gauss(self):
```

Comment:
This is random gaussian sampling on SPD? 

This code should go in the file `spd_matrices.py`, as a method of the `SPDMatrices` class called `random_gaussian`. We already have `random_uniform` there.

Reply:
I added a Gaussian sampler for SPD matrices in `spd_matrices.py`. I also added one for rotations in `special_orthogonal.py`, although it should only be viable in dimension 2 (I inserted an `assert` function to make sure of that). I didn't take the responsibility of subclassing SO(n).